lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -193,6 +213,23 @@ export class ClientCredentialClient extends BaseClient {
                 headers,
                 thumbprint
             );
+
+            // check if a new token is being requested because the cached token needs to be refreshed, not because it's expired
+            if (accessTokenCacheKey) {
+                // check if ESTS is down (status code 429 or 5xx)
+                if ((response.status === 429) || (response.status >= 500)) {
```

Comment:
Yes, this is good. I'd put it in a private method. Can we extend this condition to cover Http exceptions such as the ones you get there's no internet on the box ? I believe those exceptions do not result in http responses at all.